### PR TITLE
docs(parquet): add docstring examples for to_parquet incl. partitioning

### PR DESCRIPTION
**Pull request by gforsyth**
Following up on some comments in #5532 

Skipping the doctests that generate output since they don't clean up after themselves, but I've verified that they run.

### CODE REVIEW

File: ibis/backends/duckdb/__init__.py [hunk 1]
```diff
@@ -584,6 +591,22 @@ def to_parquet(
...
+        >>> penguins = ibis.examples.penguins.fetch()
+        >>> con = ibis.get_backend(penguins)
+        >>> con.to_parquet(penguins, "penguins.parquet")
+
+        Write out an expression to a hive-partitioned parquet file.
+
+        >>> import ibis
+        >>> penguins = ibis.examples.penguins.fetch()
+        >>> con = ibis.get_backend(penguins)
+        >>> con.to_parquet(penguins, "penguins_hive_dir", partition_by=("year"))
```
Comment (cpcloud) [inline ibis/backends/duckdb/__init__.py:609]: Are the parens around `"year"` intentional?

File: ibis/expr/types/core.py [hunk 2]
```diff
@@ -426,6 +426,22 @@ def to_parquet(
...
+
+        >>> import ibis
+        >>> penguins = ibis.examples.penguins.fetch()
+        >>> penguins.to_parquet("penguins.parquet")  # doctest: +SKIP
+
+        Write out an expression to a hive-partitioned parquet file.
+
+        >>> import ibis
+        >>> penguins = ibis.examples.penguins.fetch()
+        >>> penguins.to_parquet("penguins_hive_dir", partition_by=("year"))  # doctest: +SKIP
```
Comment (cpcloud) [inline ibis/expr/types/core.py:442]: Same question re parens.

**Review by cpcloud — APPROVED** [review]
Micronits around parens, otherwise LGTM. Ship it!

File: ibis/backends/duckdb/__init__.py (hunk 1 above)
Comment (gforsyth) [reply ibis/backends/duckdb/__init__.py:609]: yes-ish?  They're required if you partition on more than one column, but aren't required for a single-column

Comment (gforsyth) [reply ibis/backends/duckdb/__init__.py:609]: I'll add examples of both

Comment (cpcloud) [reply ibis/backends/duckdb/__init__.py]: Hm? Does it need a comma?

```
>>> ("year") == "year"
True
```

Comment (gforsyth) [reply ibis/backends/duckdb/__init__.py]: no, `duckdb` barfs if you pass in a 1-tuple

**Comment by ogrisel** [comment]
Thank you so much!

**Pull request merged**